docs/state_encryption.md

```diff
@@ -0,0 +1,333 @@
+
+```hcl2
+terraform {
+  encryption {
+    key_provider "static" "foo" {
+      key = "6f6f706830656f67686f6834616872756f3751756165686565796f6f72653169"
+    }
+  }
+}
+```
+
+**Note:** the user is responsible for keeping this key safe and follow disaster recovery best practices. OpenTofu is a read-only consumer for the key the user provided and will not perform tasks on the key management system like key rotation.
+
+The user also has to specify at least one encryption method referencing the key provider. This encryption method determines how the encryption takes place. It is the user's responsibility to make sure that they provided a key that is suitable for the encryption method.
+
+```hcl2
+terraform {
+  encryption {
+    //...
+    method "aes_gcm" "bar" {
+      key_provider = key_provider.static.foo
+    }
+  }
+}
+```
+
+Finally, the user must reference the method for use in their state file, plan file, etc. This enables creating a different configuration for different purposes.
+
+```hcl2
+terraform {
+  encryption {
+    //...
+    statefile {
+      method = method.aes_gcm.abc
+    }
+    planfile {
+      method = method.aes_gcm.cde
+    }
+    backend {
+      method = method.pbkdf2.efg
+    }
+    remote_data_sources {
+      default {
+        method = method.aes_gcm.ghi
+      }
+      remote_data_source "some_module.remote_data_source.foo" {
+        method = method.aes_gcm.ijk
+      }
+    }
+  }
+}
+```
+
+To facilitate key and method rollover, the user can specify a fallback configuration for state and plan decryption. When the user specifies a `fallback` block, `tofu` will first attempt to decrypt any state or plan it reads with the primary method and then fall back to the `fallback` method. When `tofu` writes a state or plan, it will not use the `fallback` method and always writes with the primary method.
+
+```hcl2
+terraform {
+  encryption {
+    //...
+    statefile {
+      method = method.aes_gcm.bar
+      fallback_method {
+        method = method.aes_gcm.baz
+      } 
+    }
+  }
+}
+```
+
+**Note:** The `fallback` is a block because the future goals may require adding additional options.
+
+**Note:** Multiple `fallback` blocks should not be supported because they would be detrimental to performance and encourage keeping old encryption keys in the configuration.
+
+### Composite keys
+
+When a user desires to create a composite key, such as for creating a passphrase-based derivative key or a shared custody key, they may avail themselves of a key provider that supports multiple inputs. The user can chain key providers together:
+
+```hcl2
+terraform {
+  encryption {
+    key_provider "static" "my_passphrase" {
+      key = "this is my encryption key"
+    }
+    key_provider "pbkdf2" "my_derivative_key" {
```

Comment:
If we foresee adding future encryption methods that aren't based on aes256 keys, then we need a way to indicate to pbkdf2 what the key length is. I don't think we need to implement it yet, but we should plan for it. I suppose a simple key_length field that defaults to 256 would be fine.